src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IInterpolatedStringExpression.cs

```diff
@@ -444,5 +444,890 @@ public void M(int x)
+    Predecessors: [B1]
+    Statements (1)
+        IFlowCaptureOperation: 1 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'c')
+          Value: 
+            IParameterReferenceOperation: c (OperationKind.ParameterReference, Type: System.String) (Syntax: 'c')
+
+    Next (Regular) Block[B4]
+Block[B4] - Block
+    Predecessors: [B2] [B3]
+    Statements (1)
+        IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null) (Syntax: 'b = $""{(a ? b : c)}{c}"";')
+          Expression: 
+            ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.String) (Syntax: 'b = $""{(a ? b : c)}{c}""')
+              Left: 
+                IFlowCaptureReferenceOperation: 0 (OperationKind.FlowCaptureReference, Type: System.String, IsImplicit) (Syntax: 'b')
+              Right: 
+                IInterpolatedStringOperation (OperationKind.InterpolatedString, Type: System.String) (Syntax: '$""{(a ? b : c)}{c}""')
+                  Parts(2):
+                      IInterpolationOperation (OperationKind.Interpolation, Type: null) (Syntax: '{(a ? b : c)}')
+                        Expression: 
+                          IFlowCaptureReferenceOperation: 1 (OperationKind.FlowCaptureReference, Type: System.String, IsImplicit) (Syntax: 'a ? b : c')
+                        Alignment: 
+                          null
+                        FormatString: 
+                          null
+                      IInterpolationOperation (OperationKind.Interpolation, Type: null) (Syntax: '{c}')
+                        Expression: 
+                          IParameterReferenceOperation: c (OperationKind.ParameterReference, Type: System.String) (Syntax: 'c')
+                        Alignment: 
+                          null
+                        FormatString: 
+                          null
+
+    Next (Regular) Block[B5]
+Block[B5] - Exit
+    Predecessors: [B4]
+    Statements (0)
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(source, expectedFlowGraph, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void InterpolatedStringExpression_TextAndInterpolationParts_Flow()
+        {
+            string source = @"
+using System;
+
+internal class Class
+{
+    public void M(bool a, string b, string c)
+    /*<bind>*/{
+        b = $""String1 {(a ? b : c)} and String2 {(a ? b : c)}"";
+    }/*</bind>*/
+}
+";
+            string expectedFlowGraph = @"
+Block[B0] - Entry
+    Statements (0)
+    Next (Regular) Block[B1]
+Block[B1] - Block
+    Predecessors: [B0]
+    Statements (1)
+        IFlowCaptureOperation: 0 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'b')
+          Value: 
+            IParameterReferenceOperation: b (OperationKind.ParameterReference, Type: System.String) (Syntax: 'b')
```

Comment:
>IParameterReferenceOperation: b (OperationKind.ParameterReference, Type: System.String) (Syntax: 'b') [](start = 12, length = 101)

In the future, consider using distinct parameters everywhere, makes it much easier to make sense what is going on in the base-line

Reply:
Sure, thanks

---
In reply to: [185964500](https://github.com/dotnet/roslyn/pull/26545#discussion_r185964500) [](ancestors = 185964500)